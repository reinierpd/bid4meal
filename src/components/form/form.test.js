/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { shallow } from "enzyme";
import MealForm from "./form";

const meals = [
  {
    mealId: "1",
    desc: "Some meal1",
    priceRange: {
      min: 0,
      max: 20,
      jump: 5
    },
    currency: "EUR"
  },
  {
    mealId: "2",
    desc: "Another meal1",
    priceRange: {
      min: 0,
      max: 50,
      jump: 10
    },
    currency: "EUR"
  }
];

describe("MealForm", () => {
  test("should render ok", () => {
    const wrapper = shallow(
      <MealForm meals={meals} showModal={false} onHide={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
