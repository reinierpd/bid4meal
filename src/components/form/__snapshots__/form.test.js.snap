// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealForm should render ok 1`] = `
<ForwardRef(Bootstrap(Modal))
  aria-labelledby="contained-modal-title-vcenter"
  centered={true}
  onHide={[Function]}
  show={false}
  size="lg"
>
  <ForwardRef(Bootstrap(ModalHeader))
    closeButton={true}
  >
    <ForwardRef(Bootstrap(ModalTitle))
      id="contained-modal-title-vcenter"
    >
      Add your meals
    </ForwardRef(Bootstrap(ModalTitle))>
  </ForwardRef(Bootstrap(ModalHeader))>
  <ForwardRef(Bootstrap(ModalBody))>
    <ForwardRef(Bootstrap(Form))>
      <ForwardRef(Bootstrap(FormRow))>
        <ForwardRef(Bootstrap(FormGroup))
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            }
          }
          controlId="selectedMeal"
          md="4"
        >
          <ForwardRef(ContextTransform)>
            Select your meal
          </ForwardRef(ContextTransform)>
          <ForwardRef(ContextTransform)
            as="select"
            onChange={[Function]}
          >
            <option
              key="1"
              value={0}
            >
              Some meal1
            </option>
            <option
              key="2"
              value={1}
            >
              Another meal1
            </option>
          </ForwardRef(ContextTransform)>
        </ForwardRef(Bootstrap(FormGroup))>
        <ForwardRef(Bootstrap(FormGroup))
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            }
          }
          controlId="amount"
          md={
            Object {
              "offset": 1,
              "span": 5,
            }
          }
        >
          <ForwardRef(ContextTransform)>
            Choose price (
            EUR
            )
          </ForwardRef(ContextTransform)>
          <ComponentEnhancer(undefined)
            activeDotStyle={Object {}}
            className=""
            defaultValue={0}
            disabled={false}
            dotStyle={Object {}}
            dots={false}
            handle={[Function]}
            handleStyle={
              Array [
                Object {},
              ]
            }
            included={true}
            marks={
              Object {
                "0": 0,
                "10": 10,
                "15": 15,
                "20": 20,
                "5": 5,
              }
            }
            max={20}
            min={0}
            onAfterChange={[Function]}
            onBeforeChange={[Function]}
            onChange={[Function]}
            prefixCls="rc-slider"
            railStyle={Object {}}
            step={5}
            trackStyle={
              Array [
                Object {},
              ]
            }
            vertical={false}
          />
        </ForwardRef(Bootstrap(FormGroup))>
      </ForwardRef(Bootstrap(FormRow))>
    </ForwardRef(Bootstrap(Form))>
  </ForwardRef(Bootstrap(ModalBody))>
  <ForwardRef(Bootstrap(ModalFooter))>
    <ForwardRef(Bootstrap(Button))
      onClick={[Function]}
      variant="outline-secondary"
    >
      Close
    </ForwardRef(Bootstrap(Button))>
    <ForwardRef(Bootstrap(Button))
      onClick={[Function]}
      variant="primary"
    >
      Add
    </ForwardRef(Bootstrap(Button))>
  </ForwardRef(Bootstrap(ModalFooter))>
</ForwardRef(Bootstrap(Modal))>
`;
